src/datasets/translation_datasets.py

```diff
@@ -18,16 +23,19 @@ def __init__(
         self.lang2label: dict = {}
         self.label2lang: dict = {}
         label = 0
-        for pair in tqdm(self.pairs):
+        for pair in self.pairs:
             for lang in pair.split("-"):
                 if lang not in self.lang2label:
                     self.lang2label[lang] = label
                     self.label2lang[label] = lang
                     label += 1
-            dataset = load_dataset("allenai/nllb", pair, split="train", streaming=True)
-            self.datasets[pair] = iter(dataset)
+
             self.probas.append(probas[pair] ** k)
 
+        load_dataset("allenai/nllb", self.pairs[0], split="train", streaming=True)
+        dataset_pairs = p_map(get_loaded_dataset, self.pairs)
```

Comment:
I don't like using `p_map` or any other not standard libraries. We have very common case that can be solved without additional packages.

What you need is to use `Pool` from `multiprocessing` and `imap` method in it:
```python
def load_dataset_fn(pair_name):
    return  load_dataset("allenai/nllb", pair_name, split="train", streaming=True)

dataset_pairs = [load_dataset_fn(self.pairs[0])]
with Pool(cpu_count()) as pool:
    results = pool.imap(load_dataset_fn, self.pairs[1:], chunksize=1)
    for result in tqdm(results):
        dataset_pairs.append(result)
```